ibis/backends/snowflake/__init__.py

```diff
@@ -228,6 +231,53 @@ def normalize_name(name):
         self.con.dialect.normalize_name = normalize_name
         return res
 
+    def _compile_scalar_udf(self, udf_node: ops.ScalarUDFNode):
+        dialect = self.con.dialect
+        name = type(udf_node).__name__
+        signature = ", ".join(
+            f"{name} {sa.types.to_instance(to_sqla_type(dialect, arg.output_dtype)).compile(dialect=dialect)}"
+            for name, arg in zip(udf_node.argnames, udf_node.args)
+        )
+        snowflake_type = sa.types.to_instance(
+            to_sqla_type(dialect, udf_node.output_dtype)
+        ).compile(dialect=dialect)
+        source = textwrap.dedent(inspect.getsource(udf_node.__func__)).strip()
+        source = "\n".join(
+            line for line in source.splitlines() if not line.startswith("@")
+        )
+
+        res = f"""\
+CREATE OR REPLACE TEMPORARY FUNCTION {name}({signature})
+RETURNS {snowflake_type}
+LANGUAGE python
+RUNTIME_VERSION = '3.8'
+HANDLER = '{name}'
+AS $$
+{source}
```

Comment:
For snowflake's batch API we'll need to produce slightly different code, that looks like this:

```
...
HANDLER = 'wrapper'
...
AS $$

import _snowflake
import pandas as pd

{source}

@_snowflake.vectorized(input=pd.DataFrame)
def wrapper(df):
    return {name}(*(col for _, col in df.items()))
$$
```

This is so that a UDF's signature is compatible with both scalar and batched.